File: frontend/src/api/transfer.ts
import request from '@/utils/request'
import type { ApiResponse, PageResponse } from './types'

export interface TransferData {
  id?: number
  empId: number
  transferType: number
  fromDeptId?: number
  toDeptId?: number
  fromPositionId?: number
  toPositionId?: number
  fromSalary?: number
  toSalary?: number
  transferDate: string
  reason?: string
  remark?: string
  status: number
}

export function getTransferPage(params: any) {
  return request.get<ApiResponse<PageResponse<TransferData>>>('/transfer/page', { params })
}

export function createTransfer(data: TransferData) {
  return request.post<ApiResponse>('/transfer', data)
}

export function updateTransfer(data: TransferData) {
  return request.put<ApiResponse>('/transfer', data)
}

export function approveTransfer(id: number, status: number) {
  return request.put<ApiResponse>(`/transfer/approve/${id}`, null, { params: { status } })
}

export function deleteTransfer(id: number) {
  return request.delete<ApiResponse>(`/transfer/${id}`)
}

File: backend/src/main/java/com/hrms/module/transfer/entity/Transfer.java
package com.hrms.module.transfer.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("emp_transfer")
public class Transfer {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long empId;
    /** 异动类型(1晋升 2平调 3降职 4转岗 5其他) */
    private Integer transferType;
    private Long fromDeptId;
    private Long toDeptId;
    private Long fromPositionId;
    private Long toPositionId;
    private BigDecimal fromSalary;
    private BigDecimal toSalary;
    private LocalDate transferDate;
    private String reason;
    private String remark;
    /** 状态(0待审批 1已通过 2已驳回) */
    private Integer status;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
    @TableLogic
    private Integer isDeleted;
}
